File: common/src/main/java/com/ultreon/mods/advanceddebug/client/menu/pages/WorldInfoPage.java
package com.ultreon.mods.advanceddebug.client.menu.pages;

import com.ultreon.mods.advanceddebug.api.client.menu.DebugPage;
import com.ultreon.mods.advanceddebug.api.client.menu.IDebugRenderContext;
import com.ultreon.mods.advanceddebug.api.common.Formatted;
import net.minecraft.client.Minecraft;
import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.client.multiplayer.ClientLevel;
import net.minecraft.client.multiplayer.ServerData;
import net.minecraft.client.server.IntegratedServer;
import net.minecraft.core.Vec3i;
import org.jetbrains.annotations.NotNull;

import static net.minecraft.ChatFormatting.GOLD;

public class WorldInfoPage extends DebugPage {
    public WorldInfoPage() {
    }

    @Override
    public void render(@NotNull GuiGraphics gfx, IDebugRenderContext ctx) {
        if (Minecraft.getInstance().level != null) {
            ClientLevel.ClientLevelData dimensionInfo = Minecraft.getInstance().level.getLevelData();

            IntegratedServer server = Minecraft.getInstance().getSingleplayerServer();
            ServerData currentServer = Minecraft.getInstance().getCurrentServer();
            if (server != null) {
                ctx.left("Singleplayer");
                ctx.left("World Name", server.getWorldData().getLevelName());
                ctx.left();
            }

            if (currentServer != null) {
                ctx.left("Multiplayer");
                ctx.left("Server Name", currentServer.name);
                ctx.left();
            }

            ctx.left("Misc");
            ctx.left("Spawn Angle", dimensionInfo.getSpawnAngle());
            ctx.left("Difficulty", dimensionInfo.getDifficulty());
            ctx.left("Spawn", new Vec3i(dimensionInfo.getXSpawn(), dimensionInfo.getYSpawn(), dimensionInfo.getZSpawn()));
            ctx.left("Weather", new Formatted(GOLD + (dimensionInfo.isRaining() ? (dimensionInfo.isThundering() ? "storm" : "rain") : "clear")));
            ctx.left();

            ctx.right("Flags");
            ctx.right("Difficulty Locked", dimensionInfo.isDifficultyLocked());
            ctx.right("Hardcore", dimensionInfo.isHardcore());
            ctx.right("Raining", dimensionInfo.isRaining());
            ctx.right("Thundering", dimensionInfo.isThundering());
            ctx.right();
        }
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/client/menu/pages/WorldInfoPage.kt
package com.ultreon.mods.advanceddebug.client.menu.pages

import com.mojang.blaze3d.vertex.PoseStack
import com.ultreon.mods.advanceddebug.api.client.menu.DebugPage
import com.ultreon.mods.advanceddebug.api.client.menu.IDebugRenderContext
import com.ultreon.mods.advanceddebug.api.common.Formatted
import net.minecraft.ChatFormatting
import net.minecraft.client.Minecraft
import net.minecraft.core.Vec3i

class WorldInfoPage(modId: String, name: String) : DebugPage(modId, name) {
    override fun render(poseStack: PoseStack, ctx: IDebugRenderContext) {
        if (Minecraft.getInstance().level != null) {
            val dimensionInfo = Minecraft.getInstance().level!!.levelData
            val server = Minecraft.getInstance().singleplayerServer
            if (server != null) {
                ctx.left("Singleplayer")
                ctx.left("World Name", server.worldData.levelName)
                ctx.left()
            }
            ctx.left("Misc")
            ctx.left("Spawn Angle", dimensionInfo.spawnAngle)
            ctx.left("Difficulty", dimensionInfo.difficulty)
            ctx.left("Spawn", Vec3i(dimensionInfo.xSpawn, dimensionInfo.ySpawn, dimensionInfo.zSpawn))
            ctx.left(
                "Weather",
                Formatted(ChatFormatting.GOLD.toString() + if (dimensionInfo.isRaining) (if (dimensionInfo.isThundering) "storm" else "rain") else "clear")
            )
            ctx.left()
            ctx.right("Flags")
            ctx.right("Difficulty Locked", dimensionInfo.isDifficultyLocked)
            ctx.right("Hardcore", dimensionInfo.isHardcore)
            ctx.right("Raining", dimensionInfo.isRaining)
            ctx.right("Thundering", dimensionInfo.isThundering)
            ctx.right()
        }
    }
}